app/pages/patient/patientinfo.js

```diff
@@ -399,6 +423,22 @@ var PatientInfo = React.createClass({
       return 'Diagnosis date needs to be a valid date';
     }
 
+    currentDateObj = currentDateObj || new Date();
+    var birthdayDateObj = sundial.parseFromFormat(birthday, FORM_DATE_FORMAT);
+    var diagnosisDateObj = sundial.parseFromFormat(diagnosisDate, FORM_DATE_FORMAT);
+
+    if (birthdayDateObj > currentDateObj) {
```

Comment:
Good catch!

Datetimes and timestamps are certainly interesting animals! There are a number of cases that could be problematic/work in ways that are unexpected in this area when you don't give the user explicit control over the timezone in which they are speaking. E.g.

The problem you identify.
Creating your profile in one timezone, when you were born in another (In the US, born in UK). 
Entering a the birthday of a new born in another timezone ahead of UTC.

I think what you suggest as option 2 is the best route to go. Even though the birth date and diagnosis dates are stored in UTC it restricts acceptable values based on the current date in the current time zone.

I'll update accordingly now.